src/dom-parsing/parsingAlgorithms.ts

```diff
@@ -979,12 +974,10 @@ export function parseXml(
 			);
 		}
 
+		expansionGuard.exit();
```

Comment:
The `enter` and `exit` calls seem to be out of balance. The call to `enter` is done conditionally, however `exit` is called on every loop iteration, which can even cause `depth` to go below zero.

Reply:
The enter always corresponds to a "push" on the entityContext linked list - the exit happens when removing an item from that same list. The bug here is that the exit happens exactly one time too many - for the initial item in that linked list. I'll move it into the if below to fix that.